### PR TITLE
Drop explicit requeues for OpenStackAnsibleEE

### PR DESCRIPTION
**Pull request by slagle**
- Move nova deployment execution to deployment.Deploy
- Drop explicit requeues for OpenStackAnsibleEE

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -40,13 +41,15 @@ func Deploy(
 	ctx context.Context,
 	helper *helper.Helper,
 	obj client.Object,
+	nodes *dataplanev1beta1.OpenStackDataPlaneNodeList,
```
Comment (fao89) [inline pkg/deployment/deployment.go:44]: sweet!

**Comment by fao89** [comment]
+2

File: pkg/deployment/deployment.go [hunk 2]
```diff
@@ -56,6 +59,8 @@ func Deploy(
 	var deployName string
 	var deployLabel string
 
+	logger := helper.GetLogger()
```
Comment (stuggi) [inline pkg/deployment/deployment.go:62]: redundant? above there is already `log`

File: controllers/openstackdataplanerole_controller.go [hunk 3]
```diff
@@ -227,57 +226,11 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 				err.Error()))
 			return ctrl.Result{}, err
 		}
-		if result.RequeueAfter > 0 {
+		if deployResult != nil {
+			result = *deployResult
 			return result, nil
 		}
 
-		// Call DeployNovaExternalCompute individually for each node
```
Comment (fultonj) [inline controllers/openstackdataplanerole_controller.go:234]: ack, remove creating the NovaExternalCompute CR from within the controller code and create it instead in deployment.go

**Review by fultonj — CHANGES_REQUESTED** [review]
I tested this locally in my environment and it works correctly. 

I request however, that the redundant variable `logger` be removed since we have `log`. Thanks.

File: pkg/deployment/deployment.go (hunk 2 above)
Comment (slagle) [reply pkg/deployment/deployment.go]: Done.

File: pkg/deployment/deployment.go [hunk 4]
```diff
@@ -209,7 +224,7 @@ func Deploy(
 		}
 	}
 	if !haveCephSecret {
-		helper.GetLogger().Info("Skipping execution of Ansible for ConfigureCephClient because extraMounts does not have an extraVolType of Ceph.")
+		logger.Info("Skipping execution of Ansible for ConfigureCephClient because extraMounts does not have an extraVolType of Ceph.")
```
[the remark this answers is not in the document]
Comment (slagle) [reply pkg/deployment/deployment.go]: #
Done.

File: pkg/deployment/deployment.go [hunk 5]
```diff
@@ -316,11 +337,60 @@ func Deploy(
...
+			obj,
+			sshKeySecret,
+			nodeConfigMapName,
+			status,
+			aeeSpec)
+		if err != nil {
+			novaErrors = append(novaErrors, err)
+		}
+		novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
+		logger.Info("Nova Status", "NovaExternalCompute", node.Name, "IsReady", novaExternalCompute.IsReady())
```
[the remark this answers is not in the document]
Comment (slagle) [reply pkg/deployment/deployment.go]: Done.

File: pkg/deployment/deployment.go [hunk 6]
```diff
@@ -316,11 +337,60 @@ func Deploy(
...
+			errStr = fmt.Sprintf("%s: %s", errStr, err.Error())
+		}
+		err = errors.New(errStr)
+		return &ctrl.Result{}, err
+	}
+
+	// Return when any condition is not ready, otherwise set the role as
+	// deployed.
+	if len(novaReadyConditionsTrue) < len(nodes.Items) {
+		logger.Info("Not all NovaExternalCompute ReadyConditions are true.")
```
[the remark this answers is not in the document]
Comment (slagle) [reply pkg/deployment/deployment.go]: Done.

**Review by fultonj — APPROVED** [review]
/lgtm Thanks!

**Pull request merged**